From 78306f65db7caeea2d3cb93b768232689fec3200 Mon Sep 17 00:00:00 2001
From: David Ogbeide <Davidboyowa.Ogbeide@amd.com>
Date: Mon, 31 Aug 2015 10:26:54 -0400
Subject: [PATCH 1702/2094] drm/radeon: add missing register and macro defs

-Unable to perform ops on mmSPI_ARB_PRIORITY register from
radeon driver files due to missing definitions and macros.

-Add the required reg, field, shift, and reg-op macro
definitions to allow full reg/field manipulation.

BUG: KFD-15

Change-Id: I866e45080a5b6de01838bf0f4d0b10931640b3ef
Signed-off-by: David Ogbeide <davidboyowa.ogbeide@amd.com>
Reviewed-by: Adrian Edwards <adrian.edwards@amd.com>
Reviewed-by: Bala Kumar <bala.kumar@amd.com>
---
 drivers/gpu/drm/radeon/cikd.h   | 10 ++++++++++
 drivers/gpu/drm/radeon/radeon.h | 10 ++++++++++
 2 files changed, 20 insertions(+)

diff --git a/drivers/gpu/drm/radeon/cikd.h b/drivers/gpu/drm/radeon/cikd.h
index a4a81da..e3ee553 100644
--- a/drivers/gpu/drm/radeon/cikd.h
+++ b/drivers/gpu/drm/radeon/cikd.h
@@ -2181,4 +2181,14 @@
 #define NUM_REQUESTS_AT_ERR(x)				((x) << 10)
 #define NUM_REQUESTS_AT_ERR_MASK			(0x1F << 10)
 
+#define mmSPI_ARB_PRIORITY           (0x31c0 << 2)
+#define SPI_ARB_PRIORITY__PIPE_ORDER_TS0_MASK 0x7
+#define SPI_ARB_PRIORITY__PIPE_ORDER_TS0__SHIFT 0x0
+#define SPI_ARB_PRIORITY__PIPE_ORDER_TS1_MASK 0x38
+#define SPI_ARB_PRIORITY__PIPE_ORDER_TS1__SHIFT 0x3
+#define SPI_ARB_PRIORITY__PIPE_ORDER_TS2_MASK 0x1c0
+#define SPI_ARB_PRIORITY__PIPE_ORDER_TS2__SHIFT 0x6
+#define SPI_ARB_PRIORITY__PIPE_ORDER_TS3_MASK 0xe00
+#define SPI_ARB_PRIORITY__PIPE_ORDER_TS3__SHIFT 0x9
+
 #endif
diff --git a/drivers/gpu/drm/radeon/radeon.h b/drivers/gpu/drm/radeon/radeon.h
index 60896d5..736cf53 100644
--- a/drivers/gpu/drm/radeon/radeon.h
+++ b/drivers/gpu/drm/radeon/radeon.h
@@ -2600,6 +2600,16 @@ static inline struct radeon_fence *to_radeon_fence(struct fence *f)
 #define RDOORBELL32(index) cik_mm_rdoorbell(rdev, (index))
 #define WDOORBELL32(index, v) cik_mm_wdoorbell(rdev, (index), (v))
 
+#define REG_FIELD_SHIFT(reg, field) reg##__##field##__SHIFT
+#define REG_FIELD_MASK(reg, field) reg##__##field##_MASK
+
+#define REG_SET_FIELD(orig_val, reg, field, field_val)			\
+	(((orig_val) & ~REG_FIELD_MASK(reg, field)) |			\
+	 (REG_FIELD_MASK(reg, field) & ((field_val) << REG_FIELD_SHIFT(reg, field))))
+
+#define REG_GET_FIELD(value, reg, field)				\
+	(((value) & REG_FIELD_MASK(reg, field)) >> REG_FIELD_SHIFT(reg, field))
+
 /*
  * Indirect registers accessors.
  * They used to be inlined, but this increases code size by ~65 kbytes.
-- 
2.7.4

